### ISSUE TITLE
More options to the add/edit cli commands

### ISSUE THREAD

**Issue by alexzanderr** [keepassxreboot/keepassxc#6794]
## Expectations
i want more features in keepassxc-cli add functionality

features that would be very useful:
* add description to entry (at the moment there is no option in CLI args)
* add advanced attributes to the entry (at the moment there is no option in CLI 
these are advances attributes
![image](https://user-images.githubusercontent.com/54772464/127844597-88bab397-791a-4b94-b619-f7729e1ecf0d.png)
* add custom icon to entry by specifying your icon path (at the moment there is no option in CLI args)
you know there is an icon section:
![image](https://user-images.githubusercontent.com/54772464/127844888-159e18e6-dcc4-4b9c-8f83-3eb39068eb13.png)
* and most important, add email as a main attribuite to the entry (at the moment there is only username, but for google, for example, you dont have username, you have an email that you use and you cant write email at username because this breaks the standard of the entry's logic)
here would be great to exist a field with email that must be completed in order to create entry
![image](https://user-images.githubusercontent.com/54772464/127845320-af7b05c1-20cb-444f-886f-13b2acdeb17c.png)

my wish is to have all the above features in the command line interface.

## What I use right now
right now i use this command to add entrys from terminal. (why i want to do this? because i can generate from code)
```shell
printf "%s\n%s" "password_for_database" "password_for_entry" | keepassxc-cli add -k "path_to_database_key" "path_to_database" --username "this is the username2" --url "this is the URL" "this is the title2" -p
```
i use printf to pipe the database's password and entry's password to the keepassxc-cli add in order to skip the prompt, because i dont want any prompts in the code while running from programming language.

## Want I wish to use
and here is what I would expect from you guys to add in the future:
```shell
printf "%s\n%s" "password_for_database" "password_for_entry" | keepassxc-cli add -k "path_to_database_key" "path_to_database" --username "this is the username2" --url "this is the URL" "this is the title2" -p --email "example@domain.com" --icon "icon_path" --description "quick description" --advanced-attribuites "{'key': 'value', 'key2': 'value2'}"
```
the only difference is:
```diff
+ --email "example@domain.com" 
+ --icon "icon_path" 
+ --description "quick description" 
+ --advanced-attribuites "{'key': 'value', 'key2': 'value2'}"
```
note that advanced attributes use key value pair in the GUI app, so makes sense in the end.

extra:
* i would `really` like to get rid of the password prompt for entry and for database when adding entry to the database
* and i would also like a --title argument because there is no arg like that and when im using --title it gives me an error

## Final and beautiful command
in the end i would imagine the command to be so:
```shell
keepassxc-cli add -k "path_to_database_key" "path_to_database" --username "this is the username2" --url "this is the URL" --title "this is the title2" --entry-password "entry's password" --database-password "database's password" --email "example@domain.com" --icon "icon_path" --description "quick description" --advanced-attribuites "{'key': 'value', 'key2': 'value2'}"
```
note that the difference is:
```diff
- printf "%s\n%s" "password_for_database" "password_for_entry" | 
+ --title "this is the title2"
- -p 
+ --entry-password "entry's password"
+ --database-password "database's password"
+ --email "example@domain.com" 
+ --icon "icon_path" 
+ --description "quick description" 
+ --advanced-attribuites "{'key': 'value', 'key2': 'value2'}"
```

**Comment by droidmonkey** [comment]
Please read the manpage, most of these actions  are done through the edit command. The entry name when you add is the title. 

Cli won't ever support icons, we are trying to remove the gui code at this time. 

Otherwise, the rest of your request is either duplicated elsewhere or won't be changed and has been discussed before.

**Comment by alexzanderr** [comment]
"The entry name when you add is the title" - i already knew that.
okey then, after closing this feature request with the speed of light, please tell me how can i edit `advanced attributes`, `notes` and `icon` using `edit` function, in order not to die stupid.

**Comment by droidmonkey** [comment]
`keepassxc-cli edit --help`

**Comment by alexzanderr** [comment]
do you think this is a joke, mr @droidmonkey  ??

**Comment by alexzanderr** [comment]
this command: `keepassxc-cli edit --help`
outputs the same thing as this commad: `keepassxc-cli add --help`

and there is no option that i requested (for example notes)

**Comment by alexzanderr** [comment]
so, tell me ... is it written in the `keepassxc-cli edit --help` how to change `notes`, `advanced attributes` for an exissting entry?
```
Usage: keepassxc-cli edit [options] database entry
Edit an entry.

Options:
  -q, --quiet                    Silence password prompt and other secondary
                                 outputs.
  -k, --key-file <path>          Key file of the database.
  --no-password                  Deactivate password key for the database.
  -y, --yubikey <slot[:serial]>  Yubikey slot and optional serial used to
                                 access the database (e.g., 1:7370001).
  -u, --username <username>      Username for the entry.
  --url <URL>                    URL for the entry.
  -p, --password-prompt          Prompt for the entry's password.
  -t, --title <title>            Title for the entry.
  -g, --generate                 Generate a password for the entry.
  -L, --length <length>          Length of the generated password
  -l, --lower                    Use lowercase characters
  -U, --upper                    Use uppercase characters
  -n, --numeric                  Use numbers
  -s, --special                  Use special characters
  -e, --extended                 Use extended ASCII
  -x, --exclude <chars>          Exclude character set
  --exclude-similar              Exclude similar looking characters
  --every-group                  Include characters from every selected group
  -h, --help                     Display this help.

Arguments:
  database                       Path of the database.
  entry                          Path of the entry to edit.
```
please, tell me mr @droidmonkey, where do you see `notes` and `advanced attributes` ?

because i dont see them, so I suggest you to take this situation with more seriousity and give me a good answer or dont answer at all, because i dont have time for you to tell me `look in the help`. I've looked before creating this issue!!

**Comment by droidmonkey** [comment]
Well shit I was thinking of the show command

**Comment by alexzanderr** [comment]
very much appreciate your return to the battle!

**Comment by louib** [comment]
@alexzanderr thx for the feedback

> add description to entry (at the moment there is no option in CLI args)

You're talking about the notes, right? I think we have the feature now, see https://github.com/keepassxreboot/keepassxc/pull/4646. Might not be available on the version that you're using.

> add advanced attributes to the entry (at the moment there is no option in CLI these are advances attributes

 You are right, we don't have that option yet. The same issue was raised here as well https://github.com/keepassxreboot/keepassxc/issues/2220. @droidmonkey we might want to consolidate both issues.
 
 > add custom icon to entry by specifying your icon path (at the moment there is no option in CLI args)

 Unless we're dealing with the standard database icons (icons supported by default by KeePassXC), I don't think we're going to add that to the CLI, because, as @droidmonkey said, we're trying to remove the GUI related code from the CLI. See https://github.com/keepassxreboot/keepassxc/issues/1714 for the reasoning behind that.
 
 > and most important, add email as a main attribuite to the entry 
 
 I think you should create a dedicated issue for that, as it requires a dedicated discussion, and it wouldn't only apply to the CLI, but would require changes to the GUI as well. Have a look at the existing issues, it might have already been proposed.
 
 > i would really like to get rid of the password prompt for entry and for database when adding entry to the database
 
 There's already an issue for allowing to pass the db password as an argument here https://github.com/keepassxreboot/keepassxc/issues/6752. Please add your grain of salt there :pray:  You might want to mention that you would like the feature for entry passwords too, not just for db passwords.

> and i would also like a --title argument because there is no arg like that and when im using --title it gives me an error

The title is actually the entry name (the last positional argument).

**Comment by alexzanderr** [comment]
> You're talking about the notes, right? I think we have the feature now, see #4646. Might not be available on the version that you're using.

what do you mean its not available for me? i have the lastest version (2.6.6), wait a second, the 2.7.0 version of cli is published? because i dont see it on the keepassxc website.

> Unless we're dealing with the standard database icons (icons supported by default by KeePassXC), I don't think we're going to add that to the CLI, because, as @droidmonkey said, we're trying to remove the GUI related code from the CLI. See #1714 for the reasoning behind that.

i agree, only using built in icons. that would be great.

> I think you should create a dedicated issue for that, as it requires a dedicated discussion, and it wouldn't only apply to the CLI, but would require changes to the GUI as well. Have a look at the existing issues, it might have already been proposed.

i will, because `email` its a very important attribute, nowadays most accounts use the email as main authentication factor 
* you get verification codes on email
* you receive notifications about account on email
and I, in my most account, have email more `frequently` than username. this doesnt mean username should be removed. no, ofc. i consider `email` and `username` should be equally valued.

> There's already an issue for allowing to pass the db password as an argument here #6752. Please add your grain of salt there pray You might want to mention that you would like the feature for entry passwords too, not just for db passwords.

no, no, thats the thing i didnt wanted. i dont want to `pipe` the password, i just want to `pass` the password as `argument` to keepassxc-cli like so:
```diff
keepassxc-cli add \
+ --entry-password "some password for entry" \
+ --database-password "database's password" \
... (more args)
```
this is the exact principle on which the key file for database is inserted. it has a `-k` argument to specifiy the key file path for the database. i want exactly the same principle, but just for passwords because i dont want to pipe anymore.

> The title is actually the entry name (the last positional argument).

yes, i already knew that the last positional argument is the title. but i consdier it inconvenient, its not specific and not readable. it would be beautiful and easy to have:
```diff
keepassxc-cli add  ... (arguments) --title "--title before the title its very easy to read and very specific" ... (more arguments)
```

**Comment by louib** [comment]
@alexzanderr 

> what do you mean its not available for me? i have the lastest version (2.6.6), wait a second, the 2.7.0 version of cli is published? because i dont see it on the keepassxc website.

Yes, please try with the latest version of the cli.


> i agree, only using built in icons. that would be great.

I created issue https://github.com/keepassxreboot/keepassxc/issues/8242 to track the issue.

> no, no, thats the thing i didnt wanted. i dont want to pipe the password, i just want to pass the password as argument to keepassxc-cli like so:

Then please create a dedicated issue for that, so that we can discuss this issue separately.

> yes, i already knew that the last positional argument is the title. but i consdier it inconvenient, its not specific and not readable. it would be beautiful and easy to have:

Again, please create a dedicated issue for that :pray:

**Comment by louib** [comment]
Closing in favor of tracking the raised issues in separate tickets.
